include/seqan3/io/stream/iterator.hpp

```diff
@@ -0,0 +1,60 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides seqan3::ostream and seqan3::ostreambuf iterator.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
+
+#ifdef __cpp_lib_ranges
+#include <iterator>
```

Comment:
you need to include the header before checking the macro, otherwise it may not be defined.

Reply:
Ok, I included <version> which will provide all feature test macros.

Reply:
version is c++20, it broke your builds... You can just include iterator undconditionally...

Reply:
Yeah, I just realized that. Haven’t read the notes. Could have been tagged for c++20 more clearly on cppreference though 😒